src/app/home/room-transaction/room-transaction.component.ts

```diff
@@ -256,4 +305,189 @@ export class RoomTransactionComponent implements OnInit {
       }
     })
   }
+
+  onToggle(e: MatSlideToggleChange){
+    this.foodDataSource = null;
+    if(e.checked){
+      this.SlideText="Table Transaction/s"
+      this.transactionType =false;
+      this.selectedRoom = false;
+      this.getTable();
+    }else{
+      this.selectedTable=false;
+      this.SlideText="Room Transaction/s"
+      this.transactionType=true;
+      this.getRoomList();
+    }
+
+
+  }
+
+  getRoomList() {
+    this.blockUI.start('Loading...')
+    const roomParams = {
+      limit: this.limit,
+      skip: this.skip
+    }
+    this.roomService.getRoomList(roomParams).subscribe(
+      (result) => {
+        if (result && result.data) {
+          this.totalLength = result.totalCount
+          this.roomTranDataSource = result.data
+        } else {
+          this.blockUI.stop()
+        }
+        this.blockUI.stop()
+      },
+      (error) => {
+        this.blockUI.stop()
+      }
+    )
+  }
+
+  
+  getTable() {
+    this.blockUI.start('Loading...')
+    const paginationParams = {
+      limit: this.limit,
+      skip: this.skip
+    }
+    this.tableService.getTableList(paginationParams).subscribe(
+      (result) => {
+        const arr = []
+        if (result && result.data) {
+          result.data.map((table) => {
+            this.totalLength = result.totalCount
+            arr.push({
+              table_number:table.table_number,
+              table_id:table.id,
+              status: table.food_orders[0].invoice_id == null ? 'Due' : 'Paid',
+              callFrom:'tableTransaction'
+            });
+          });
+          this.tableDataSource = new MatTableDataSource(arr);
+          this.blockUI.stop();
+        } else {
+          this.blockUI.stop()
+        }
+        this.blockUI.stop()
+      },
+      (error) => {
+        this.blockUI.stop()
+      }
+    )
+  }
+
+  getTotalFoodCost(){
+    this.foodTotal = 0;
+    if(this.selectedFoodDetail){
+      this.foodTotal = this.selectedFoodDetail.map(food => food.food_items.price *food.quantity).reduce((acc, value) => acc + value, 0);
+    }
+    return parseFloat(this.foodTotal);
+  }
+
+  getGrandTotal(){
+    this.grandTotal =0;
+    if(this.roomDataSource.amount && this.selectedRoom){
+      this.grandTotal =parseFloat(this.foodTotal) + parseFloat(this.roomDataSource.amount);
+    }
+     return this.grandTotal;
+
+  }
+
+  displaySelectedRoomDetail(data){
+   data.middle_name= data.middle_name?data.middle_name:''
```

Comment:
```suggestion
   data.middle_name= data.middle_name?data.middle_name:''
```
same variable is assigned